packages/driver/src/cy/commands/navigation.ts

```diff
@@ -850,6 +954,12 @@ export default (Commands, Cypress, cy, state, config) => {
 
           knownCommandCausedInstability = true
 
+          // if this is a cross origin spec bridge, we need to tell the primary to change
+          // the AUT iframe since we don't have access to it
+          if (Cypress.isCrossOriginSpecBridge) {
+            return Cypress.specBridgeCommunicator.toPrimary('visit:url', { url })
```

Comment:
What are we gaining by sending this event via the specBridgeCommunicator vs directly calling `$utils.iframeSrc(Cypress.$autIframe, url)`? Snapshots?

Reply:
if this is a spec bridge we cannot communicate with the top frame because it will be a cross origin frame, so instead we send a message through the specBridgeCommunicator for the top frame to make this change.